ndcube/ndcube.py

```diff
@@ -543,10 +544,11 @@ def _get_crop_by_values_item(self, *points, units=None, wcs=None):
                             f"index: {i}; coord type: {type(value)}; unit: {unit}")
                     points[i][j] = u.Quantity(value, unit=unit)
                 elif value is not None:
```

Comment:
```suggestion
                if value is not None:
```
This conversion needs to be done even if `value_is_float` as there's no guarantee that the user will enter the values in the `wcs_axis_units`.